### DISCUSSION TITLE
Provide a way to build android apps without proprietary dependencies

### DISCUSSION THREAD

**Discussion by gilbsgilbs** [ionic-team/capacitor#3443] [Ideas]
Currently, it is not possible to build capacitor apps for android as free software because the android module has transitive dependencies to [firebase-messaging and play-services-location](https://github.com/ionic-team/capacitor/blob/a4539be88ba939e793466badf7b2899294c05246/android/capacitor/build.gradle#L69). This is a blocker for those of us who want to produce free builds of their apps and more specifically who want to make them available on [F-Droid](https://f-droid.org/). F-Droid requires completely free and reproducible builds which means that not only the final APK shouldn't include proprietary dependencies, but also the whole build process must not involve proprietary software at all. Therefore, making firebase and gms `compileOnly` dependencies is not a solution.

Those dependencies could be avoided by providing a build flag (e.g. from an environment variable) or just by making firebase and location services a separate module (which I think is cleaner, but would probably be a breaking change). A nice-to-have feature would be allowing users to create multiple variants of their app with different dependencies, e.g. with or without firebase. Something similar to what the android gradle plugin allows with [product flavors](https://developer.android.com/studio/build/build-variants#product-flavors).

Thanks.

**Comment by jcesarmobile** [comment]
Capacitor 3 is moving the plugins out, so that won't be a problem anymore. But of course, you won't be able to use the plugins that require those dependencies.

**Reply by gilbsgilbs** [reply]
That't great, I didn't know about it. Then the idea would be about having an optional dependencies mechanism, but that is less of an issue as it is easier to deal with those at project level. Thanks a bunch for your quick answer!
